### ISSUE TITLE
How can I define a schedule that lists specific date ranges

### ISSUE THREAD

**Issue by felixfbecker** [renovatebot/config-help#94]
We have 4 week iterations at Sourcegraph, which means they don't always align with the start of months. I want Renovate to only run in the first week of every iteration (planning/tech debt week). I would like to hard-code the date ranges for one year.

I've tried coding this into a schedule (testing with the underlying node module that does the parsing) but I cannot figure out how to get a proper schedule like this. Is it possible?

**Comment by rarkins** [comment]
It's probably not possible with the current implementation, however I don't have anything philosophically against it if `later` can support it. Or alternatively supporting a whole other schedule syntax if it doesn't.

Apart from checking the hours of the day, the way we checks days themselves is:
- if a day of the week (i.e. Monday - Sunday) is defined then check if current day matches
- if a day of the month (i.e. 1st - 31st) is defined then check if that matches today

So therefore it doesn't support absolute date ranges like you need.

If you want to configure weeklong periods every four weeks for a year, then that means you want a way to configure 13 different ranges, for example? And you would be ok to configure them as whole days?

Schedule already is an array with logical "OR", so that construct is compatible with your requirements. So then either:
- We invent a new non-`later` string format which is just "start - stop" where from and to are ISO8601 date formats
- We allow overloading of schedule to be either a string or an object, and the object can have start and stop fields

**Comment by felixfbecker** [comment]
I would be happy with _any_ way to accomplish this, but I understand that Renovate should design this feature carefully, as configuring schedule is a core feature. So far my experience with defining schedules has been terrible - the `later` library has this "natural language" syntax that, of course, does not fully support natural language, and I ended up installing the library and debugging countless attempts following this grammar that would usually error until I finally found something that works.

![image](https://user-images.githubusercontent.com/10532611/45156986-0cc7a000-b1e0-11e8-86fc-7904f1ef5ece.png)

The library is unmaintained btw: https://github.com/bunkat/later#this-project-is-no-longer-being-maintained-it-should-be-treated-as-sample-code-on-one-technique-to-calculate-future-instances-of-complex-schedules-if-you-are-looking-to-fire-a-task-every-1-minute-use-a-cron-scheduler-instead

I would indeed prefer something standard & clearly defined over a fancy natural language range. One idea: could Renovate just support standard [ISO 8601 Time Interval](https://en.wikipedia.org/wiki/ISO_8601#Time_intervals) strings? They also support recurring intervals. I.e. when Renovate runs, it would check if the current date is within any of the intervals defined in the array, with something like https://www.npmjs.com/package/iso8601-interval-recurrence. If the string doesn't parse as a ISO 8601 interval, it would fall back to natural language with `later`. This is pretty easy to determine since ISO 8601 strings are well defined.

An object structure would also work for my use case, but I can see that becoming inflexible (e.g. what if someone wants to restrict it to only run at night?).
